tools/autograd/gen_python_functions.py

```diff
@@ -245,190 +239,157 @@ def group_declarations_by_op_name(declarations):
 
 def create_python_bindings(python_functions, is_python_method, module):
     """Generates Python bindings to ATen functions"""
-    py_methods = []
-    py_method_defs = []
-    py_forwards = []
+    py_methods: List[str] = []
+    py_method_defs: List[str] = []
+    py_forwards: List[str] = []
 
     for name in sorted(python_functions.keys()):
-        overload_decls = python_functions[name]
-
-        for declaration in overload_decls:
-            # TODO: change all methods to directly process python signatures instead of decls.
-            declaration['python_signature'] = decl_to_python_signature(declaration, method=is_python_method)
-            declaration['native_function'] = decl_to_native_function(declaration)
-
-        py_methods.append(method_impl(name, overload_decls, is_python_method, module))
-        py_method_defs.append(method_def(name, overload_decls, is_python_method, module))
-        py_forwards.extend(forward_decls(name, overload_decls, is_python_method, module))
+        overloads = tuple(decl_to_signature_function_pair(decl, method=is_python_method)
+                          for decl in python_functions[name])
+        noarg = len(overloads) == 1 and overloads[0].signature.arguments_count() == 0
```

Comment:
Note to self: wondering what a noarg is and why we have to compute here, rather than inside the functions.

Reply:
no particular reason - just that this way we can avoid passing 'overloads' to 'method_def' / 'forward_decls' because this bit is what all they need.
I've changed it back to passing overloads instead to make it more uniform.